### PR TITLE
function96 keyboard added

### PR DESCRIPTION
**Pull request by JuanTapMan**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
Could you please run through the PR checklist and update accordingly? You have marked it as done but there are a few things still outstanding.

**Comment by JuanTapMan** [comment]
Updated to PR Checklist specifications

File: keyboards/function96/config.h [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (noroadsleft) [inline keyboards/function96/config.h:25]:
```suggestion
#define PRODUCT_ID 0x672A
```

File: keyboards/function96/config.h [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
+// /*== customize breathing effect ==*/
+//   /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//   #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+//   /*==== use exp() and sin() ====*/
+//   #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//   #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+// #endif
+
```
Comment (noroadsleft) [inline keyboards/function96/config.h:87]:
```suggestion
```

File: keyboards/function96/config.h [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
```
Comment (noroadsleft) [inline keyboards/function96/config.h:249]:
```suggestion
```

File: keyboards/function96/function96.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+#define LAYOUT_all( \
```
Comment (noroadsleft) [inline keyboards/function96/function96.h:30]: As there is only one layout possible, this should be named `LAYOUT`.

```suggestion
#define LAYOUT( \
```

File: keyboards/function96/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/function96/keymaps/default/keymap.c:19]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/function96/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (noroadsleft) [inline keyboards/function96/keymaps/default/keymap.c:58]:
```suggestion
```

File: keyboards/function96/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    [0] = LAYOUT_all(
+        KC_ESC ,   KC_F1,   KC_F2,  KC_F3, KC_F4,  KC_F5,   KC_F6, KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12, KC_PSCR,  KC_DEL, KC_MPRV, KC_MNXT, KC_VOLU, KC_VOLD, \
+        KC_GRV ,    KC_1,    KC_2,   KC_3,  KC_4,   KC_5,    KC_6,  KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,  KC_EQL, KC_BSPC, KC_MPLY, KC_PSLS, KC_PAST, KC_PMNS, \
+        KC_TAB ,    KC_Q,    KC_W,   KC_E,  KC_R,   KC_T,    KC_Y,  KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,   KC_P7,   KC_P8,   KC_P9, KC_PPLS, \
+        KC_CAPS,    KC_A,    KC_S,   KC_D,  KC_F,   KC_G,    KC_H,  KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT,  KC_ENT,   KC_P4,   KC_P5,   KC_P6, \
+        KC_LSFT,    KC_Z,    KC_X,   KC_C,  KC_V,   KC_B,    KC_N,  KC_M, KC_COMM,  KC_DOT, KC_SLSH, KC_RSFT,   KC_UP,   KC_P1,   KC_P2,   KC_P3, KC_PENT, \
+        KC_LCTL, KC_LWIN, KC_LALT, KC_SPC, MO(1), KC_SPC, KC_RALT, MO(2), KC_LEFT, KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT
+      ),
+
+    [1] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/function96/keymaps/default/keymap.c:28]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/function96/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+    [2] = LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/function96/keymaps/default/keymap.c:36]:
```suggestion
    [2] = LAYOUT(
```

File: keyboards/function96/readme.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+Function96
+===
```
Comment (noroadsleft) [inline keyboards/function96/readme.md:2]:
```suggestion
# Function96
```

File: keyboards/function96/readme.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
+# Function96
+===
```
Comment (noroadsleft) [inline keyboards/function96/readme.md:2]:
```suggestion
```

File: keyboards/function96/readme.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+This Function96 is a passion project of JTM. It's meant to give the user the maximum usability of the space allocated by the keyboard while also giving the option of using the numpad.
+
+Keyboard Maintainer: The QMK Community  
+Hardware Supported: Function96
+Hardware Availability: Limited Groupbuy that ended on 8/14/2020, https://www.reddit.com/r/mechmarket/comments/i7wwh4/gb_function96_prototype/
```
Comment (noroadsleft) [inline keyboards/function96/readme.md:8]:
```suggestion
* Keyboard Maintainer: The QMK Community  
* Hardware Supported: Function96
* Hardware Availability: Limited Groupbuy that ended on 8/14/2020, https://www.reddit.com/r/mechmarket/comments/i7wwh4/gb_function96_prototype/
```

**Comment by JuanTapMan** [comment]
> 
> 
> GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10024/files).

Oh, whoa, that's a good tip.



> 
> 
> The layout macros and keymap could use some white space cleanup, but I'd prefer to leave that for a later PR.

Working on that one, I'll get that done in a hot sec.

**Comment by JuanTapMan** [comment]
Also, sorry if any of this stuff is super wrong or bad practice. This is my first time trying anything like this or using Github.

File: keyboards/function96/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
+      ),
+    [2] = LAYOUT(
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
```
Comment (fauxpark) [inline keyboards/function96/keymaps/default/keymap.c:41]:
```suggestion
        KC_ESC ,   KC_F1,   KC_F2,  KC_F3, KC_F4,  KC_F5,   KC_F6, KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12, KC_PSCR,  KC_DEL, KC_MPRV, KC_MNXT, KC_VOLU, KC_VOLD,
        KC_GRV ,    KC_1,    KC_2,   KC_3,  KC_4,   KC_5,    KC_6,  KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,  KC_EQL, KC_BSPC, KC_MPLY, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB ,    KC_Q,    KC_W,   KC_E,  KC_R,   KC_T,    KC_Y,  KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,   KC_P7,   KC_P8,   KC_P9, KC_PPLS,
        KC_CAPS,    KC_A,    KC_S,   KC_D,  KC_F,   KC_G,    KC_H,  KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT,  KC_ENT,   KC_P4,   KC_P5,   KC_P6,
        KC_LSFT,    KC_Z,    KC_X,   KC_C,  KC_V,   KC_B,    KC_N,  KC_M, KC_COMM,  KC_DOT, KC_SLSH, KC_RSFT,   KC_UP,   KC_P1,   KC_P2,   KC_P3, KC_PENT,
        KC_LCTL, KC_LWIN, KC_LALT, KC_SPC, MO(1), KC_SPC, KC_RALT, MO(2), KC_LEFT, KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT
      ),

    [1] = LAYOUT(
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______
      ),
    [2] = LAYOUT(
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
        _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
```

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**